File: Lyft Backend Software Engineering/tires/tires_type/carrigan_tires.py
from tires.tires import Tires

# CarriganTires class inherits from Tires and represents Carrigan tires.
class CarriganTires(Tires):
    # Constructor initializes CarriganTires with tire wear data and service threshold.
    def __init__(self, tire_wear):
        self.tire_wear = tire_wear
        self.tire_need_to_service = 0.9

    # needs_service method checks if any tire needs service based on wear threshold.
    def needs_service(self):
        for tire in self.tire_wear:
            if tire >= self.tire_need_to_service:
                return True
        return False

File: Softstraw Back-End Engineer/tires/tires_type/carrigan_tires.java
package tires.tires_type;

import tires.Tires;

// CarriganTires class inherits from Tires and represents Carrigan tires.
public class CarriganTires extends Tires {

    // Field to store tire wear data and service threshold.
    private double[] tireWear;
    private final double tireNeedToService = 0.9;

    // Constructor initializes CarriganTires with tire wear data.
    public CarriganTires(double[] tireWear) {
        this.tireWear = tireWear;
    }

    // needsService method checks if any tire needs service based on the wear threshold.
    @Override
    public boolean needsService() {
        for (double tire : tireWear) {
            if (tire >= tireNeedToService) {
                return true;
            }
        }
        return false;
    }
}
